cairo/ethereum/cancun/trie.cairo

```diff
@@ -803,12 +855,42 @@ func _get_branches(obj: MappingBytesBytes, level: Uint) -> (TupleMappingBytesByt
     return (branches_tuple, value);
 }
 
+func _get_preimage_for_key{poseidon_ptr: PoseidonBuiltin*}(
```

Comment:
why go through the entire dict for every lookup
if possible, we should index the preimage dict with associated hashes to speed up lookups